### DISCUSSION TITLE
CLI hangs if DB can't connect, unable to CTRL+C

### DISCUSSION THREAD

**Discussion by Kangaroux** [sqlalchemy/alembic#1001] [Usage Questions]
**Describe the bug**
If the CLI can't connect to the DB because of a firewall or other issue, it never seems to timeout, and you can't stop it with CTRL+C or CTRL+D. 

**Expected behavior**
Include a timeout (5s?) and/or allow CTRL+C to kill the CLI.

**To Reproduce**
In the DB conn str, set the database host to something like `example.com`

**Versions.**
 - OS: Alpine (Docker)
 - Python: 3.9.9
 - Alembic: 1.7.5
 - SQLAlchemy: 1.4.29
 - Database: Postgres
 - DBAPI:

**Have a nice day!**

**Comment by zzzeek** [comment]
this is not a SQLAlchemy issue as SQLAlchemy does not intercept the KeyboardInterrupt error.   you likely want to consult the issue tracker for the DBAPI you're using (not stated here) and/or consult the Python mailing list.
